Add fuse support for batchnorm with affine=False

sstsai-adl opened the pull request:
Summary: When batchnorm affine is set to false, weight and bias is set to None, which is not supported in this case. Added a fix to set weights to 1 and bias to 0 if they are not set.

Test Plan: Add unit test for testing fusing conv, batchnorm where batchnorm is in affine=False mode.

Differential Revision: D23977080

z-a-f reviewed (approved):
LGTM! Thank you!

vadimkantorov commented on torch/nn/utils/fusion.py:
```diff
@@ -16,6 +16,10 @@ def fuse_conv_bn_eval(conv, bn):
     if conv_b is None:
         conv_b = bn_rm.new_zeros(bn_rm.shape)
+    if bn_w is None:
+        bn_w = torch.ones(bn_rm.shape)
```
should it instead be `torch.ones_like(bn_rm)` to preserve device? or at least `torch.ones(bn_rm.shape, device = bn_rm.device)`?

sstsai-adl committed:
Add fuse support for batchnorm with affine=False (#45474)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/45474

When batchnorm affine is set to false, weight and bias is set to None, which is not supported in this case. Added a fix to set weights to 1 and bias to 0 if they are not set.

Test Plan: Add unit test for testing fusing conv, batchnorm where batchnorm is in affine=False mode.

Reviewed By: z-a-f

Differential Revision: D23977080

fbshipit-source-id: cc11797d227049aaaec01a125966dfebdaf8b176

vadimkantorov commented on torch/nn/utils/fusion.py:
```diff
@@ -16,6 +16,10 @@ def fuse_conv_bn_eval(conv, bn):
 def fuse_conv_bn_weights(conv_w, conv_b, bn_rm, bn_rv, bn_eps, bn_w, bn_b):
     if conv_b is None:
         conv_b = bn_rm.new_zeros(bn_rm.shape)
```
Should it not be `conv_b = torch.zeros_like(bn_rm)` here as well?

sstsai-adl replied:
Will create another PR for that refactor.
